File: src/me/xerces/eventbus/test/EventBusTest.java
package me.xerces.eventbus.test;

import me.xerces.eventbus.IEventManager;
import me.xerces.eventbus.SimpleEventManager;
import me.xerces.eventbus.annotation.EventHandler;
import me.xerces.eventbus.test.event.EventTest;
import me.xerces.eventbus.test.event.EventTestTwo;

import java.text.DecimalFormat;
import java.util.concurrent.TimeUnit;

/**
 * @author Xerces
 * @since 24/01/2015
 */
public class EventBusTest {

    public static void main(String[] args)
    {
        new EventBusTest();
    }

    public EventBusTest()
    {
        IEventManager eventManager = new SimpleEventManager();
        eventManager.addEventListener(new EventListenerOne());
        eventManager.addEventListener(new EventListenerTwo());
        int eventsToFire = 1000000;
        long lastMS = System.nanoTime();
        for(int i = 0; i < eventsToFire; i++)
        {
            eventManager.fireEvent(new EventTest());
            eventManager.fireEvent(new EventTestTwo());
        }
        lastMS = System.nanoTime() - lastMS;
        System.out.println(String.format("Firing 1 million events took %dms", TimeUnit.MILLISECONDS.convert(lastMS, TimeUnit.NANOSECONDS)));
    }


}

File: src/main/java/dev/throwouterror/eventbus/test/EventBusTest.kt
package dev.throwouterror.eventbus.test

import dev.throwouterror.eventbus.SimpleEventBus
import dev.throwouterror.eventbus.event.Event
import dev.throwouterror.eventbus.observeEvent
import java.util.concurrent.TimeUnit

class EventBusTest {
    companion object {
        @JvmStatic
        fun main() {
            EventBusTest()
        }
    }

    init {
        val eventManager = SimpleEventBus()
        val eventsToFire = 1000000
        var lastMS = System.nanoTime()
        for (i in 0 until eventsToFire) {
            eventManager.fireEvent(Event("test"))
        }
        lastMS = System.nanoTime() - lastMS
        println(String.format("Firing 1 million events took %dms", TimeUnit.MILLISECONDS.convert(lastMS, TimeUnit.NANOSECONDS)))
        eventManager.observeEvent("test2").subscribe {
            println("Receieved event ${it.type}")
        }
        eventManager.fireEvent(Event("test2"))
    }
}
